lib/api/controller/auth.js

```diff
@@ -188,9 +189,74 @@ class AuthController extends NativeController {
         request.context.token);
     }
 
+    if ( global.kuzzle.config.security.supportCookieAuthentication
+      && request.input.args.cookieOnly
+    ) {
+      request.setResult(null,
+        {
+          headers: {
+            'Set-Cookie': Cookie.serialize(
+              '__Host-authToken',
+              '',
+              {
+                httpOnly: true,
+                path: '/',
+                sameSite:'none',
+                secure: true,
+              }
+            )
+          }
+        });
+    }
+
     return {acknowledged: true};
   }
 
+  // Used to send the Token using different ways when in cookieOnly mode. (DRY)
+  async _sendToken(token, request) {
+    // Only if the support of Browser Cookie as Authentication Token is enabled
+    // otherwise we should send a normal response because
+    // even if the SDK / Browser can handle the cookie,
+    // Kuzzle would not be capable of doing anything with it
+    if (global.kuzzle.config.security.supportCookieAuthentication
+      && request.input.args.cookieOnly
+    ) {
+      // Here we are not sending jwt when cookieOnly is set to true
+      // This allow us to detect if kuzzle does support cookie as jwt directly from the SDK
+      // if the response contains the jwt this means that the configuration [security.supportCookieAuthentication] is set to false
+      // or that the version of kuzzle doesn't support the feature Browser Cookie as Jwt
+
+      request.setResult(null,
+        {
+          headers: {
+            'Set-Cookie': Cookie.serialize(
+              '__Host-authToken',
+              token.jwt,
+              {
+                expires: new Date(token.expiresAt),
+                httpOnly: true,
+                path: '/',
+                sameSite:'none',
+                secure: true,
```

Comment:
As already demanded/discussed, can you explain why you chose to use the `secure: true` property?
What about when in dev mode where **https** is not enabled?

[Forbid non-https cookies with Secure flag](https://blog.dareboost.com/en/2019/03/secure-cookies-secure-httponly-flags/#forbid-to-use-a-cookie-without-https-thanks-to-the-secure-flag)
[Other issues when using secure + httpOnly](https://webmasters.stackexchange.com/questions/53178/what-are-the-problems-of-httponly-and-secure-cookies#:~:text=The%20problem%20with%20secure%20cookies,can%20save%20your%20server%20resources.)

Reply:
Secure cookie cannot be delivered when requests are not made with the https protocol, but they can be delivered to localhost even if it's not https.

I think it's a bigger concern to send token through http than not being able to use cookies for your website when in http

Reply:
You right, I just saw that I didn't push something, because I stashed it, now `sameSite` is set to `'strict'` and `secure` has been removed allowing us to use the cookie even in an http context, which was not the case before.

`sameSite: 'strict'` is a better choice since the browser will not send cookies when the request is not made directly from the same domain as the cookie was obtained, this helps us to avoid CSRF attacks

Reply:
I agree with @Leodau , this feature should work in both HTTP and HTTPS environments

Reply:
This is the case, as I mentionned above, cookies works both in HTTP and HTTPS